numpy/_core/src/umath/string_buffer.h

```diff
@@ -1505,4 +1505,81 @@ string_expandtabs(Buffer<enc> buf, npy_int64 tabsize, Buffer<enc> out)
 }
 
 
+enum class JUSTPOSITION {
+    CENTER, LEFT, RIGHT
+};
+
+template <ENCODING enc>
+static inline npy_intp
+string_pad(Buffer<enc> buf, npy_int64 width, Buffer<enc> fill, JUSTPOSITION pos, Buffer<enc> out)
+{
+    size_t finalwidth = width > 0 ? width : 0;
+    size_t len = buf.num_codepoints();
+    if (len >= finalwidth) {
+        buf.buffer_memcpy(out, len);
```

Comment:
Should this just copy `finalwidth` elements?

Reply:
[CPython specifies](https://docs.python.org/3/library/stdtypes.html#str.center) that the original string is returned when `width <= len`.

Reply:
OK, I think I now got it. I was misled by the examples in the docstring, which showed truncated strings.